### PR TITLE
HHH-13523 - StatementPreparerImpl should not need to retrieve the JDDCService as often

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-13523

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/StatementPreparerImpl.java [hunk 1]
```diff
@@ -199,11 +205,7 @@ private void setStatementTimeout(PreparedStatement preparedStatement) throws SQL
 	}
 
 	private JdbcServices getJdbcService() {
```
Comment (Sanne) [inline hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/StatementPreparerImpl.java]: do we still need the `getJdbcService()` method ?

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/StatementPreparerImpl.java]: probably not let me check

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/StatementPreparerImpl.java [hunk 2]
```diff
@@ -37,6 +38,11 @@
 	 */
 	StatementPreparerImpl(JdbcCoordinatorImpl jdbcCoordinator) {
 		this.jdbcCoordinator = jdbcCoordinator;
+		this.jdbcServices = jdbcCoordinator
```
Comment (Sanne) [inline hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/StatementPreparerImpl.java]: this still looks quite "heavy" . I see `JdbcCoordinatorImpl` is the only place invoking this ctor, and it also retrives the `jdbcServices` in its own constructor.. keep it and pass it on directly?

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/StatementPreparerImpl.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/StatementPreparerImpl.java]: I'll remove it and add another little change

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/StatementPreparerImpl.java (hunk 2 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/StatementPreparerImpl.java]: I'll do

**Comment by Sanne** [comment]
rebased & merged

**Pull request closed without merging**